### ISSUE TITLE
Make \ref in doxygen comments a clickable link to the symbol

### ISSUE THREAD

**Issue by spoorcc** [microsoft/vscode-cpptools#11131]
### Feature Request

In Doxygen you can make a reference to a symbol using the [`\ref`](https://www.doxygen.nl/manual/commands.html#cmdref) command. It would be great if the referenced symbol could become a clickable link so you can jump to the implementation.

**Comment by spoorcc** [comment]
Something like this? https://vscode-api.js.org/interfaces/vscode.DocumentLinkProvider.html#provideDocumentLinks

**Comment by michelleangela** [comment]
Yes, the [DocumentLinkProvider](https://code.visualstudio.com/api/references/vscode-api#DocumentLinkProvider) is probably what would be needed.

**Comment by KKghub** [comment]
I see that if we write link to the file with prefix `file:///` in comment, VSCode shows it as clickable link. and it supports relative links as well. For example:

<img width="341" alt="image" src="https://github.com/microsoft/vscode-cpptools/assets/25261656/ee0297d4-4f55-4b7d-8430-11181277b6e2">

It would be nice if we do not need to prefix it with `file:///` to make it clickable, maybe similar to how markdowns detect it? 
```md
[Test](./test.cpp)
```
